spec/inputs/xmpp_spec.rb

```diff
@@ -1 +1,106 @@
+# encoding: utf-8
 require "logstash/devutils/rspec/spec_helper"
+require 'logstash/inputs/xmpp'
+
+module Jabber
+  class ClientMock
+    def initialize(jid)
+      @jid = jid
+      @message_block = ->(msg) { puts "!ERROR! ----> Should not see #{msg}"}
+    end
+    def connect(*) true; end
+    def auth(*) true; end
+    def send(*) true; end
+    def close(*) true; end
+    def add_presence_callback(*) true; end
+    def add_message_callback(*args, &block)
+      @message_block = block
+    end
+    def inject_message(msg)
+      @message_block.call(msg)
+    end
+  end
+
+  class MucClientMock
+    def initialize(client)
+      @message_block = ->(time, from, body) { puts "!!!!! ----> Should not see #{time}, #{from}, #{body}"}
+    end
+    def join(rm)
+    end
+    def on_message(&block)
+      @message_block = block
+    end
+    def inject_message(time, from, body)
+      @message_block.call(time, from, body)
+    end
+  end
+
+  class MessageMock
+    From = Struct.new(:node, :domain, :resource)
+    attr_reader :body, :from
+    def initialize(str)
+      @body = str
+      @from = From.new('bar','domain.org','chat')
+    end
+  end
+end
+
+describe LogStash::Inputs::Xmpp do
+  let(:rooms) { ['logstash', 'kibana'] }
+  let(:config) do
+    {
+      'user' => 'foo@domain.org/chat',
+      'password' => 'bAr',
+      'rooms' => rooms
+    }
+  end
+
+  before do
+    allow(Jabber::Client).to receive(:new) do |jid|
+      Jabber::ClientMock.new(jid)
+    end
+
+    allow(Jabber::MUC::SimpleMUCClient).to receive(:new) do |client|
+      Jabber::MucClientMock.new(client)
+    end
+  end
+
+  context 'when running' do
+    let(:queue) { [] }
+    let(:msg)   { 'Hello Logstash'}
+
+    subject { described_class.new(config) }
```

Comment:
I know is standard and I know what I means, just found so many people confused with it, on the other side it requires you to remember what you defined in the upper describe section, so is also error prone in that point too.

Both options are ok and do their work, just a matter usage.